src/install/lockfile.zig

```diff
@@ -2677,6 +2677,7 @@ pub const Package = extern struct {
                         total_dependencies_count += @truncate(u32, arr.items.len);
                     },
                     .e_object => |obj| {
+                        if (group.behavior.isWorkspace()) return error.InvalidPackageJSON;
```

Comment:
maybe this should have a more specific error

> "Workspaces expects an array of package names as strings, like this:\nexample"

Reply:
Done in https://github.com/oven-sh/bun/pull/2074/commits/6164cd62fd812921115df4e7864b3a718304fc31